tests/unit/core/test_message.py

```diff
@@ -91,6 +91,20 @@ def test_pack_unpack():
     assert str(struct) == str(unpacked)
     assert struct == unpacked
 
+    # test pack/unpack for dict object with default serializtion as JSON
+    dictionary = {"key": 0.1233444444, "hodor": "hold the door", "default": 0}
+    message = Message(content=dictionary)
+    assert message.content_type == ContentType.JSON
+    unpacked = message.unpack()
+    assert dictionary == unpacked
+
+    # test pack/unpack for dict object with protobuf content-type
+    message = Message()
```

Comment:
What if I pack a dictionary that fits well on a protobuf schema, specify the content type as protobuf, and than try to unpack with the correct schema:

```python
>>> from is_msgs.common_pb2 import Pose
>>> from is_wire.core import Message, ContentType
>>> p_dict={'position': {'x': 1.0, 'y': 0.0, 'z': 0.0}}
>>> msg = Message(content=p_dict, content_type=ContentType.PROTOBUF)
>>> msg.unpack(Pose)
position {
}
>>> msg.unpack()
{'position': {'x': 1.0, 'y': 0.0, 'z': 0.0}}
```
Not sure if it's possible to handle with this situation, because the parse process is not able to verify if the schema is correct, it just tries to fit on the given one.